sight_service/worklist_scheduler_opt.py

```diff
@@ -61,53 +58,52 @@ def launch(self,
   @overrides
   def propose_action(
       self, request: service_pb2.ProposeActionRequest
-  ) -> service_pb2.ProposeActionResponse:
+    ) -> service_pb2.ProposeActionResponse:
     # print('request in propose actions: ', request)
 
     attributes = param_proto_to_dict(request.attributes)
     action_attrs = param_proto_to_dict(request.action_attrs)
 
-    with self.pending_lock.gen_wlock():
-      self.pending_samples[self.unique_id] = [action_attrs, attributes]
+    unique_id = self.queue.push_message(message={
+        "action_atrrs": action_attrs,
+        "attributes": attributes
+    })
 
-    # print('self.pending_samples : ',
-    #       self.pending_samples)
-    # print('self.active_samples : ',
-    #       self.active_samples)
-    # print('self.completed_samples : ',
-    #       self.completed_samples)
-    print('self.unique_id : ', self.unique_id)
+    logging.info("self.queue => %s", self.queue)
 
-    # Create response
-    response = service_pb2.ProposeActionResponse(action_id=self.unique_id)
-    self.unique_id += 1
+    response = service_pb2.ProposeActionResponse(action_id=unique_id)
     return response
 
   @overrides
   def GetOutcome(
       self,
       request: service_pb2.GetOutcomeRequest) -> service_pb2.GetOutcomeResponse:
-    # print('self.pending_samples : ',
-    #       self.pending_samples)
-    # print('self.active_samples : ',
-    #       self.active_samples)
-    # print('self.completed_samples : ',
-    #       self.completed_samples)
-    with self.completed_lock.gen_rlock():
-      completed_samples = self.completed_samples
-    with self.pending_lock.gen_rlock():
-      pending_samples = self.pending_samples
-    with self.active_lock.gen_rlock():
-      active_samples = self.active_samples
+
+    logging.info('self.queue => %s', self.queue)
+
+    all_messages = self.queue.get_all_messages()
 
     response = service_pb2.GetOutcomeResponse()
-    if (request.unique_ids):
+    if not request.unique_ids:
+      for sample_id in all_messages["completed"]:
+        sample_details = all_messages["completed"][sample_id]
+        outcome = response.outcome.add()
+        outcome.action_id = sample_id
+        outcome.status = service_pb2.GetOutcomeResponse.Outcome.Status.COMPLETED
+        outcome.reward = sample_details['reward']
+        outcome.action_atrrs.extend(
```

Comment:
action_attrs

Reply:
It's present in the previous code's RPC , will update in next PR